File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.java
class Solution {
    public boolean solution(int x) {
        boolean answer = true;
        String [] num = Integer.toString(x).split("");
        int number = 0 ;
        for(int i = 0 ; i < num.length ; i++){
            number += Integer.parseInt(num[i]);
        }
        
        if(x % number != 0){
            answer = false;
        }
        
        return answer;
    }
}

File: C#/프로그래머스/1/12947. 하샤드 수/하샤드 수.cs
public class Solution {
    public bool solution(int x) {
        bool answer = true;
                    int check2 = 0;
        int num = x;
            while (x > 0)
            {
                check2 += x % 10;
                x /= 10;
            }
            if (num % check2 != 0) answer = false;

            return answer;
        return answer;
    }
}